Arrow function annotated with `v128` that returns null crashes the compiler

<!--
 Thanks for submitting an issue to AssemblyScript! Please take a moment to
 read the contributing guidelines linked below to get off to a good start 🙂
-->

## Code

```ts
(): v128 => null
// this also works
(): dataref => null
```

[Try it in the playground](https://www.assemblyscript.org/editor.html#IyFvcHRpbWl6ZT1zcGVlZCZydW50aW1lPXN0dWIKKCk6IHYxMjggPT4gbnVsbAojIWh0bWwKPHRleHRhcmVhIGlkPSJvdXRwdXQiIHN0eWxlPSJoZWlnaHQ6IDEwMCU7IHdpZHRoOiAxMDAlIiByZWFkb25seT48L3RleHRhcmVhPgo8c2NyaXB0Pgpsb2FkZXIuaW5zdGFudGlhdGUobW9kdWxlX3dhc20sIHsgLyogaW1wb3J0cyAqLyB9KQogIC50aGVuKCh7IGV4cG9ydHMgfSkgPT4gewogICAgY29uc3Qgb3V0cHV0ID0gZG9jdW1lbnQuZ2V0RWxlbWVudEJ5SWQoJ291dHB1dCcpCiAgICBmb3IgKGxldCBpID0gMDsgaSA8PSAxMDsgKytpKSB7CiAgICAgIG91dHB1dC52YWx1ZSArPSBgZmliKCR7aX0pID0gJHtleHBvcnRzLmZpYihpKX1cbmAKICAgIH0KICB9KQo8L3NjcmlwdD4=)

## Stack trace

```
▌ Whoops, the AssemblyScript compiler has crashed during compile :-(
▌ 
▌ Here is the stack trace hinting at the problem, perhaps it's useful?
▌ 
▌ AssertionError: assertion failed
▌     at assert (D:\assemblyscript\std\portable\index.js:200:11)
▌     at Module.flatten (D:\assemblyscript\src\module.ts:1454:7)
▌     at Compiler.compileFunction (D:\assemblyscript\src\compiler.ts:1571:16)
▌     at Compiler.compileFunctionExpression (D:\assemblyscript\src\compiler.ts:7572:25)
▌     at Compiler.compileExpression (D:\assemblyscript\src\compiler.ts:3502:21)
▌     at Compiler.compileExpressionStatement (D:\assemblyscript\src\compiler.ts:2521:17)
▌     at Compiler.compileStatement (D:\assemblyscript\src\compiler.ts:2213:21)
▌     at Compiler.compileTopLevelStatement (D:\assemblyscript\src\compiler.ts:2175:25)
▌     at Compiler.compileFile (D:\assemblyscript\src\compiler.ts:1111:12)
▌     at Compiler.compile (D:\assemblyscript\src\compiler.ts:486:14)
▌     at Object.compile (D:\assemblyscript\src\index.ts:280:32)
▌     at D:\assemblyscript\cli\asc.js:902:31
▌     at measure (D:\assemblyscript\cli\asc.js:1409:3)
▌     at Object.main (D:\assemblyscript\cli\asc.js:900:24)
▌     at runTest (D:\assemblyscript\tests\compiler.js:178:7)
▌ 
▌ If it refers to the dist files, try to 'npm install source-map-support' and
▌ run again, which should then show the actual code location in the sources.
▌ 
▌ If you see where the error is, feel free to send us a pull request. If not,
▌ please let us know: https://github.com/AssemblyScript/assemblyscript/issues
▌ 
▌ Thank you!
```

--- a/src/module.ts
+++ b/src/module.ts
@@ -1452,7 +1452,10 @@ export class Module {
         }
       }
       let singleType = getExpressionType(single);
-      assert(singleType == TypeRef.Unreachable || singleType == type);
+      if (singleType != TypeRef.Unreachable && singleType != type) {
+        // can happen when there was a diagnostic prior
+        return this.unreachable();
+      }
       return single;
     }
     return this.block(null, stmts, type);

